# Исследование надежности заемщиков

*Данные*

Входные данные от банка — статистика о платёжеспособности клиентов.

*Цель исследования*

Выяснить влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. 

*Описание данных*

- children — количество детей в семье
- days_employed — общий трудовой стаж в днях
- dob_years — возраст клиента в годах
- education — уровень образования клиента
- education_id — идентификатор уровня образования
- family_status — семейное положение
- family_status_id — идентификатор семейного положения
- gender — пол клиента
- income_type — тип занятости
- debt — имел ли задолженность по возврату кредитов
- total_income — ежемесячный доход
- purpose — цель получения кредита

*Дополнительная информация*

Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

*Ход исследования*

1. Обзор данных. На этом этапе будет проведены обзор и оценка качества данных.
2. Предобработка данных. На этом этапе будет осуществлена проверка данных на ошибки и оценка их влияние на исследование. По возможности исправление критичных ошибок данных.
3. Проведение исследовательского анализа для для определения факторов, кторые влияют на погашения кредита в срок.

## Обзор данных
Создадим окружение для работы.

In [1]:
import pandas as pd

Ознакомимся с данными.

In [2]:
try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

In [3]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


### Итоги обзора данных

Мы имеем дело с данными, которые содержат 21525 наблюдений с описанием 12 признаков. Выявлены пропуски в данных: total_income, days_employed. Видим, что некоторые данные записаны как строчным, так и заглавным шрифтом. Наблюдаются аномальные значения.

Далее будет проведена предобработка данных.

## Предобработка данных
### Удаление пропусков
Рассмотрим количество пропущенных значений для каждого столбца.

In [5]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

В двух столбцах есть пропущенные значения. Один из них — days_employed. Пропуски в этом столбце мы обработаем на следующем этапе. Другой столбец с пропущенными значениями — total_income — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполним пропуски в этом столбце медианным значением по каждому типу из столбца income_type. 

In [6]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

In [7]:
data.isna().sum()# проверка заполнения пропусков

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income           0
purpose                0
dtype: int64

### Обработка аномальных значений
Отрицательное количество дней трудового стажа в столбце days_employed - является аномальным. Обработаем значения в этом столбце: заменим все отрицательные значения положительными с помощью метода abs().

In [8]:
data['days_employed'] = data['days_employed'].abs()

In [9]:
data.head()# проверка замены

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   Среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу

Для каждого типа занятости найдем медианное значение трудового стажа days_employed в днях.

In [10]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставим их как есть. Этот столбец не понадобится для исследования.

Рассмотрим перечень уникальных значений столбца children.

In [11]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

 В столбце children есть два аномальных значения. Удалим строки, в которых встречаются такие аномальные значения из датафрейма data.

In [12]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

In [13]:
data['children'].unique()# проверка удаления значений

array([1, 0, 3, 2, 4, 5])

Заполним пропуски в столбце days_employed медианными значениями по каждому типу занятости income_type.

In [14]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

In [15]:
data.isna().sum()# проверка заполнения пропусков

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

Пропуски заполнены.

### Изменение типов данных
Заменим вещественный тип данных в столбце total_income на целочисленный с помощью метода astype().

In [16]:
data['total_income'] = data['total_income'].astype(int)

In [17]:
data.info()# проверка замены типа данных

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21402 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21402 non-null  int64  
 1   days_employed     21402 non-null  float64
 2   dob_years         21402 non-null  int64  
 3   education         21402 non-null  object 
 4   education_id      21402 non-null  int64  
 5   family_status     21402 non-null  object 
 6   family_status_id  21402 non-null  int64  
 7   gender            21402 non-null  object 
 8   income_type       21402 non-null  object 
 9   debt              21402 non-null  int64  
 10  total_income      21402 non-null  int64  
 11  purpose           21402 non-null  object 
dtypes: float64(1), int64(6), object(5)
memory usage: 2.1+ MB


### Обработка дубликатов
Рассмотрим неявные дубликаты в столбце education. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведем их к нижнему регистру. Проверим остальные столбцы.

In [18]:
data['education'] = data['education'].str.lower()

In [19]:
data.duplicated().sum()

71

In [20]:
data = data.drop_duplicates()

In [21]:
data.duplicated().sum()# проверка удаления дубликатов

0

### Категоризация данных
На основании диапазонов, указанных ниже, создфдим в датафрейме data столбец total_income_category с категориями:

- 0–30000 — 'E';
- 30001–50000 — 'D';
- 50001–200000 — 'C';
- 200001–1000000 — 'B';
- 1000001 и выше — 'A'.

In [22]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [23]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

In [24]:
data.head()# проверка категоризации

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose total_income_category  
0               покупка жилья                     B  
1     приобретение автомобиля                     C  
2               покупка жилья                     C  
3  дополнительное образование                     B  
4             сыграть свадьбу                     C

Рассмотрим перечень уникальных целей взятия кредита из столбца purpose.

In [25]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

Создадим функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:

- 'операции с автомобилем',
- 'операции с недвижимостью',
- 'проведение свадьбы',
- 'получение образования'.

Например, если в столбце purpose находится подстрока 'на покупку автомобиля', то в столбце purpose_category должна появиться строка 'операции с автомобилем'.


In [26]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [27]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

In [28]:
data.head()# проверка категоризации

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose total_income_category          purpose_category  
0               покупка жилья                     B  операции с недвижимостью  
1     приобретение автомобиля                     C    операции с автомобилем  
2               покупка жилья                     C  операции с недвижимостью  
3  дополнительное образование                     B     получение образования  
4             сыграть свадьбу                     C        проведение свадьбы

### Итоги предобработки данных
Нами были заполнены пропущенные значения, обработаны аномальные значения, изменен тип данных, устранены дубликаты и проведена категоризация данных в некоторых столбцах.

Далее проведём исследовательский анализ для для определения факторов, кторые влияют на погашения кредита в срок.

## Исследовательский анализ
### Есть ли зависимость между количеством детей и возвратом кредита в срок?¶

In [29]:
children_res = data.pivot_table(index='children', values='debt', aggfunc=['count', 'sum', 'mean'])
children_res.columns = ['Количество заемщиков', 'Количество просрочек', 'Процент просрочек']
children_res['Процент просрочек'] = round(children_res['Процент просрочек'] * 100, 2)
children_res

Количество заемщиков  Количество просрочек  Процент просрочек
children                                                               
0                        14091                  1063               7.54
1                         4808                   444               9.23
2                         2052                   194               9.45
3                          330                    27               8.18
4                           41                     4               9.76
5                            9                     0               0.00

**Вывод:** 
1. Установлена зависимость: c увеличением количества детей мы видим увеличение количества просроченных задолженностей. Бездетные, как правило, реже опаздывают с внесением оплаты по кредиту, чем люди с детьми;
2. Выявлены неоднозначные данные:
- люди с 3 детьми чаще платят в срок чем люди с 1 ребенком;
- люди с 5 детьми всегда платят в срок.

Для данных категориий клиентов банка нужна большая выборка, чем та, которую мы имеем.

### Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [30]:
status_res = data.pivot_table(index='family_status', values='debt', aggfunc=['count', 'sum', 'mean'])
status_res.columns = ['Количество заемщиков', 'Количество просрочек', 'Процент просрочек']
status_res['Процент просрочек'] = round(status_res['Процент просрочек'] * 100, 2)
status_res

Количество заемщиков  Количество просрочек  \
family_status                                                       
Не женат / не замужем                  2796                   273   
в разводе                              1189                    84   
вдовец / вдова                          951                    63   
гражданский брак                       4134                   385   
женат / замужем                       12261                   927   

                       Процент просрочек  
family_status                             
Не женат / не замужем               9.76  
в разводе                           7.06  
вдовец / вдова                      6.62  
гражданский брак                    9.31  
женат / замужем                     7.56

**Вывод:** 

Установлена зависимость: 
1. Клиенты банка не состоящие в браке и клиенты, состоящие в гражданском браке, чаще опаздывают с внесением оплаты по кредиту (или не платят по кредиту);
2. Овдовевшие клиенты банка и клиенты, состоящие в разводе, чаще вовремя вносят оплату по кредиту. 

### Есть ли зависимость между уровнем дохода и возвратом кредита в срок?
Описание категорий по уровню заработка:

0–30000 — 'E';

30001–50000 — 'D';

50001–200000 — 'C';

200001–1000000 — 'B';

1000001 и выше — 'A'.

In [31]:
income_res = data.pivot_table(index='total_income_category', values='debt', aggfunc=['count', 'sum', 'mean'])
income_res.columns = ['Количество заемщиков', 'Количество просрочек', 'Процент просрочек']
income_res['Процент просрочек'] = round(income_res['Процент просрочек'] * 100, 2)
income_res

Количество заемщиков  Количество просрочек  \
total_income_category                                               
A                                        25                     2   
B                                      5014                   354   
C                                     15921                  1353   
D                                       349                    21   
E                                        22                     2   

                       Процент просрочек  
total_income_category                     
A                                   8.00  
B                                   7.06  
C                                   8.50  
D                                   6.02  
E                                   9.09

**Вывод:** 

Зависимость между уровнем дохода и возвратом кредита в срок не выявлена: уровень дохода никак не влияет на возврат кредита в срок.

### Как разные цели кредита влияют на его возврат в срок?

In [32]:
purpose_res = data.pivot_table(index='purpose_category', values='debt', aggfunc=['count', 'sum', 'mean'])
purpose_res.columns = ['Количество заемщиков', 'Количество просрочек', 'Процент просрочек']
purpose_res['Процент просрочек'] = round(purpose_res['Процент просрочек'] * 100, 2)
purpose_res

Количество заемщиков  Количество просрочек  \
purpose_category                                                       
операции с автомобилем                    4279                   400   
операции с недвижимостью                 10751                   780   
получение образования                     3988                   369   
проведение свадьбы                        2313                   183   

                          Процент просрочек  
purpose_category                             
операции с автомобилем                 9.35  
операции с недвижимостью               7.26  
получение образования                  9.25  
проведение свадьбы                     7.91

**Вывод:** 

1. Клиенты банка, взявшие кредит на операции с автомобилем и получение образования, чаще опаздывают с внесением оплаты по кредиту (или не платят по кредиту;
2. Чаще всего, клиенты банка платят в срок по кредиту, взятого на операции с недвижимостью и проведение свадьбы.

###  Возможные причины появления пропусков в исходных данных
Мы  обнаружили в столбцах days_employed и total_income пропущенные значения, причём, количество пропусков в обоих категориях совпадает. Из этого следует, что есть зависимость отработанных дней и суммарного дохода клиентов банка. Можно предположить, что отсутствие данных в описанных столбцах возникло из-за неправильной обработки полученной от клиента информации об отработанных днях и суммарном доходе.

Необходимо обратить внимание на качество входных данных и наличие обязательства указания данной информации клиентами банка и (или) обработчиком данных.

### Почему заполнили пропуски медианным значением для количественных переменных
Пропуски в значениях количественных переменных следует заполнять характерными значениями. Это возможно благодаря тому, что количественные значения можно сравнивать между собой и оценивать состояние выборки в общем и целом. Для примерной оценки типичных значений выборки было выбрано медианное значение, которое делит отсортированные по возрастанию данные на две равные части. Медианное значение является лучшим, так как распределение данных искажено и заполнение пропусков, например, средними арифмитическими значениями, будет являться ещё более грубой заменой пропусков, чем медианными значениями для данной выборки.

## Общий вывод
В результате подготовки данных к последующему анализу были выполнены:
    
1. Работа с пропусками.
В столбцах days_employed и total_income количество пропущенных значений было значительным (2174 в каждом столбце) и пропущенные значения были заменены медианными. 

    
2. Обработка аномальных значений.
В столбце days_employed у двух типов (безработные и пенсионеры) были выявлены аномально большие значения, которые не были изменены или удалены, т.к. не использовались в дальнейшем исследовании. В столбце children возраст был указан  "-1" и "20". Cтроки с данными значениями были удалены.
    
    
3. Изменение типов данных.
В столбце total_income был заменён вещественный тип данных на целочисленный.
    
    
4. Обработка дубликатов.
В столбце education было дублирование: одни и те же значения, записанные с использованием заглавных и строчных букв. Значения были приведены к единому виду - к нижнему регистру.
    
    
5. Категоризация данных.¶
Данные из столбца total_income были разбиты на категории, в результате чего был создан столбец total_income_category в котором отражены категори по уровню дохода:
- 0–30000 — 'E';
- 30001–50000 — 'D';
- 50001–200000 — 'C';
- 200001–1000000 — 'B';
- 1000001 и выше — 'A'.


    Данные из столбца purpose были разбиты на категории, в результате чего был создан столбец purpose_category, в котором отражены  категори целей взятия кредита:
    
- операции с автомобилем;
- операции с недвижимостью;
- проведение свадьбы;
- получение образования.
    

**В результате исследования зависимости между количеством детей и возвратом кредита в срок были получены следующие данные:**

In [33]:
children_res[children_res['Процент просрочек'].isin({children_res['Процент просрочек'].max(), 
                                                     children_res['Процент просрочек'].min()})]

Количество заемщиков  Количество просрочек  Процент просрочек
children                                                               
4                           41                     4               9.76
5                            9                     0               0.00

Самая успешная группа заемщиков - заемщики с пятью детьми (должников нет), наименее успешная - семьи с 2 и 4 детьми (до 9.7% оплат по кредиту не в срок).

Выявлены неоднозначные данные: люди с 3 детьми чаще платят в срок чем люди с 1 ребенком; люди с 5 детьми всегда платят в срок.

Для данных категориий клиентов банка нужна большая выборка, чем та, которую мы имеем.

**В результате исследования зависимость между семейным положением и возвратом кредита в срок были получены следующие данные:**

In [34]:
status_res[status_res['Процент просрочек'].isin({status_res['Процент просрочек'].max(),
                                                 status_res['Процент просрочек'].min()})]

Количество заемщиков  Количество просрочек  \
family_status                                                       
Не женат / не замужем                  2796                   273   
вдовец / вдова                          951                    63   

                       Процент просрочек  
family_status                             
Не женат / не замужем               9.76  
вдовец / вдова                      6.62

Самая успешная группа заемщиков - овдовевшие заемщики и заемщики, состоящие в разводе (до 6,6% оплат по кредиту не в срок), наименее успешная - не состоящие в браке и заемщики, состоящие в гражданском браке (до 9.7%).

**В результате исследования зависимость между уровнем дохода и возвратом кредита в срок были получены следующие данные:**

In [35]:
income_res[income_res['Процент просрочек'].isin({income_res['Процент просрочек'].max(),
                                                 income_res['Процент просрочек'].min()})]

Количество заемщиков  Количество просрочек  \
total_income_category                                               
D                                       349                    21   
E                                        22                     2   

                       Процент просрочек  
total_income_category                     
D                                   6.02  
E                                   9.09

Самая успешная группа заемщиков - с уровнем дохода 30001–50000 (до 6% оплат по кредиту не в срок), наименее успешная с уровнем дохода - 0–30000 (до 9% оплат по кредиту не в срок). Зависимость между уровнем дохода и возвратом кредита в срок не выявлена: уровень дохода никак не влияет на возврат кредита в срок.

**В результате исследования зависимость разных целей кредита и возвратом кредита в срок были получены следующие данные:**

In [36]:
purpose_res[purpose_res['Процент просрочек'].isin({purpose_res['Процент просрочек'].max(),
                                                   purpose_res['Процент просрочек'].min()})]

Количество заемщиков  Количество просрочек  \
purpose_category                                                       
операции с автомобилем                    4279                   400   
операции с недвижимостью                 10751                   780   

                          Процент просрочек  
purpose_category                             
операции с автомобилем                 9.35  
операции с недвижимостью               7.26

Самая успешная группа заемщиков, взявшие кредит на операции с недвижимостью и проведение свадьбы (до 7,2% оплат по кредиту не в срок), наименее успешная - взявшие кредит на операции с автомобилем и получение образования (до 9% оплат по кредиту не в срок).    


**На возврат кредита в срок влияют следующие данные о клиентах банка:**
1. Количеством детей. 
C увеличением количества детей мы видим увеличение количества просроченных задолженностей. Бездетные клиенты банка реже опаздывают с внесением оплаты по кредиту, чем люди с детьми; 
2. Семейное положение. 
Клиенты банка не состоящие в браке и клиенты, состоящие в гражданском браке,чаще опаздывают с внесением оплаты по кредиту (или не платят по кредиту).Овдовевшие клиенты банка и клиенты, состоящие в разводе, чаще вовремя вносят оплату по кредиту;
3. Цели кредита.
Клиенты банка, взявшие кредит на операции с автомобилем и получение образования, чаще опаздывают с внесением оплаты по кредиту (или не платят по кредиту). Чаще всего клиенты банка платят в срок по кредиту взятого на операции с недвижимостью и проведение свадьбы.

**На возврат кредита в срок не влияет уровнень дохода клиентов банка.**
